File: week07/whale.pl
#!/usr/bin/perl

@args=();
%individuals=();
%count=();
for $elem (@ARGV){
	while (EOF != ($input = <STDIN>)){
	#print "$input";
		my ($number, $animal) = split / /, $input, 2;
		chomp($animal);
		if (exists $count{$animal}) {
			$count{$animal} += 1;
			$individuals{"$animal"} += $number;
		} else {
			$count{"$animal"} = 1;
			$individuals{"$animal"} = $number;
		}
	}

#print the hash
#foreach $name (keys %hash) {
#$hash{$name} =~ s/,//; # remove comma prefix
#print "$name $hash{$name}\n";
#}
		if (exists $count{$elem}){
			print "$elem observations: $count{$elem} pods, $individuals{$elem} individuals\n";
		} else {
			print "$elem observations: 0 pods, 0 individuals\n";
		}
	


}

File: week06/whale.py
#!/usr/bin/python2.7

pods = dict()
individuals = dict()

import sys
import re

for line in sys.stdin:
	lineinput = line.split(" ")
	animal = lineinput[1].split("\n")
	if animal[0] in individuals:
		individuals[animal[0]] += lineinput[0]
		pods[animal[0]] += 1
	else:
		individuals[animal[0]] = lineinput[0]
		pods[animal[0]] = 1
space = " "
stuff = sys.argv
stuff.remove(sys.argv[0])
input_animal = space.join(stuff)
#print (sys.argv, "observations: ", pods[sys.argv], "pods")
if space in individuals:
	print stuff, "observations: ", pods[stuff], "pods,", individuals[stuff], "individuals"
else:

	print stuff, "observations: 0 pods, 0 individuals"
